sw/device/silicon_creator/rom/rom.c

```diff
@@ -156,9 +176,14 @@ static rom_error_t rom_init(void) {
   // Update epmp config for debug rom according to lifecycle state.
   rom_epmp_config_debug_rom(lc_state);
 
-  // Re-initialize the watchdog timer.
-  watchdog_init(lc_state);
-  SEC_MMIO_WRITE_INCREMENT(kWatchdogSecMmioInit);
+  if (launder32(waking_from_sleep) != kHardenedBoolTrue) {
+    HARDENED_CHECK_EQ(waking_from_sleep, kHardenedBoolFalse);
+    // Re-initialize the watchdog timer, if the RESET was caused by anything
+    // besides waking from low power (which would have left the watchdog in its
+    // previous configuration).
+    watchdog_init(lc_state);
```

Comment:
If we leave this as implemented here, we probably need to document what is the minimum counter configuration supported by the ROM, since we don't have any WD pet commands as of now.

Reply:
Yes, I think that will be necessary.  I assume we want to let the owner code configure watchdog, and even lock its configuration, so we will need to state that if such owner code want to make use of deep sleep, how many ticks could elapse between waking up and control being handed back to owner code (or at least how much time could pass between ROM/ROM_EXT petting the watchdog, until handing over control to owner code.)